openai-api-http/src/main/java/bg/codexio/ai/openai/api/http/DefaultOpenAIHttpExecutor.java

```diff
@@ -210,31 +210,44 @@ protected static <T> T getField(
         }
     }
 
+    public void configureMappingExternally(Consumer<ObjectMapper> mappingConsumer) {
+        mappingConsumer.accept(this.objectMapper);
+    }
+
     /**
      * {@inheritDoc}
      */
     @Override
     public O execute(I request) {
-        var httpRequest = prepareRequest(request);
+        var httpRequest = this.prepareRequest(request);
 
-        try (
-                var httpResponse = this.client.newCall(httpRequest)
-                                              .execute()
-        ) {
-            this.throwOnError(httpResponse);
+        return this.performRequestExecution(httpRequest);
+    }
 
-            return this.toResponse(httpResponse);
-        } catch (IOException e) {
-            throw new HttpCallFailedException(
-                    this.baseUrl + this.resourceUri,
-                    e
-            );
-        }
+    @Override
+    public O executeWithPathVariables(String... pathVariables) {
+        var httpRequest = this.prepareRequestWithPathVariables(pathVariables);
+
+        return this.performRequestExecution(httpRequest);
+    }
+
+    @Override
+    public O executeWithPathVariable(
```

Comment:
Can we use only the `pathVariables` plural method and just sending one thing, if only one is needed?

Also, not very sure about it, but isn't it better to have only one method for execution and it to deduct path variables from the request object? Maybe there can be a `Map` or a `List` with path variable values? Which will be empty for most of our request models of course.

Reply:
Yes, it is good approach to put it in the request object, but it is going to work only in situations where we are sending POST request ( in the methods `execute` and `executeWithPathVariable` ) and if we choose this as a solution we are going to send the POST request using only ONE method : `execute` for both request with, or without path variables. This doesn't means that the `executeWithPathVariables` is redundant, because it is doing a different kind of work. It is sending GET request, so body there is not required and path variables needs to be passed as parameter of the method. For example in the RUNS API , while retrieving a run for its status the path is looking like this : `https://api.openai.com/v1/threads/{thread_id}/runs/{run_id}`. Maybe the method `executeWithPathVariables` , can be renamed, since its actual work is just executing GET request.